### PR TITLE
fix: Add fallback schedule for non-onboarded team members

### PR DESCRIPTION
**Pull request by hariombalhara**
## What does this PR do?

Such members could be part of a team and a team event routed with those users will have no timeslots to be booked

It follows the philosophy that no prospect should go without being able to book a call. The member can set availability when they feel the schedule is wrong.

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] N/.A I have updated the developer docs in /docs if this PR makes changes that would require a [documentation change](https://cal.com/docs). If N/A, write N/A here and check the checkbox.
- [ ] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?
- Make a team event in acme org with all members toggle enabled in asssignment tab
- Invite a member to acme org(he will be automatically added to the team event)
- Try booking the team event with query param cal.routedTeamMemberIds={MEMBER_USER_ID}
- There will be no slots available earlier but now it would be bookable.

### CODE REVIEW

**Review by hbjORbj — DISMISSED** [review]

**Review by keithwillcode — DISMISSED** [review]

File: packages/trpc/server/routers/viewer/slots/util.ts [hunk 1]
```diff
@@ -917,8 +919,30 @@ export async function getAvailableSlots({ input, ctx }: GetScheduleOptions): Pro
...
+            email: host.email,
+            user: host.user.id,
+          })),
+        },
+      }
+    : null;
+
   return {
     slots: withinBoundsSlotsMappedToDate,
+    ...troubleshooterData,
```
Comment (Udit-takkar) [inline packages/trpc/server/routers/viewer/slots/util.ts:945]: when `enableTroubleshooter` is false then we can't spread `troubleshooterData`

File: packages/trpc/server/routers/viewer/slots/util.ts [hunk 2]
```diff
@@ -917,8 +919,30 @@ export async function getAvailableSlots({ input, ctx }: GetScheduleOptions): Pro
...
+            };
+          }),
+          hosts: eventHosts.map((host) => ({
+            email: host.email,
+            user: host.user.id,
+          })),
+        },
+      }
+    : null;
+
```
Comment (Udit-takkar) [inline packages/trpc/server/routers/viewer/slots/util.ts:942]:
```suggestion
  const troubleshooterData = enableTroubleshooter
    ? {
        troubleshooter: {
          // One that Salesforce asked for
          askedContactOwner: contactOwnerEmailFromInput,
          // One that we used as per Routing
          usedContactOwner: contactOwnerEmail,
          routedHosts: routedHostsWithContactOwnerAndFixedHosts.map((host) => {
            return {
              email: host.email,
              user: host.user.id,
            };
          }),
          hosts: eventHosts.map((host) => ({
            email: host.email,
            user: host.user.id,
          })),
        },
      }
    : {};

```

**Review by Udit-takkar — DISMISSED** [review]
1) Unit tests are failing

  Error: Snapshot `GET /api/slots > Success > Regular event-type > Returns and event type available slots with passed timeZone 1` 
  
  2) https://github.com/calcom/cal.com/pull/17196#discussion_r1807820377

File: packages/trpc/server/routers/viewer/slots/util.ts (hunk 1 above)
Comment (hariombalhara) [reply packages/trpc/server/routers/viewer/slots/util.ts:945]: `null` is spreadable.

File: packages/trpc/server/routers/viewer/slots/util.ts (hunk 2 above)
Comment (hariombalhara) [reply packages/trpc/server/routers/viewer/slots/util.ts:942]: Not needed,

File: apps/api/v1/pages/api/slots/_get.test.ts [hunk 3]
```diff
@@ -66,11 +66,8 @@ describe("GET /api/slots", () => {
         buildMockData();
         await handler(req, res);
         console.log({ statusCode: res._getStatusCode(), data: JSON.parse(res._getData()) });
-        expect(JSON.parse(res._getData())).toMatchInlineSnapshot(`
-          {
-            "slots": {},
```
Comment (hariombalhara) [inline apps/api/v1/pages/api/slots/_get.test.ts:71]: The intentiona seems just to be verifying `slots` prop to be there but now slots property can't be empty if the user is there.

Comment (hariombalhara) [reply apps/api/v1/pages/api/slots/_get.test.ts:71]: cc @zomars

**Review by Udit-takkar — APPROVED** [review]

File: apps/api/v1/pages/api/slots/_get.test.ts (hunk 3 above)
Comment (zomars) [reply apps/api/v1/pages/api/slots/_get.test.ts:71]: As long as there's always a slot properly even if there's no available ones were good. For preventing breaking changes @hariombalhara

**Pull request merged**